File: 2021/src/day01.rs
use crate::aoc;

pub fn part1(input: String) -> String {
	let mut prev = i32::MAX;
	let mut count = 0;

	for line in input.split('\n') {
		let i: i32 = line.parse().unwrap();

		if i > prev {
			count += 1;
		}

		prev = i;
	}

	return count.to_string()
}


pub fn part2(input: String) -> String {
	let mut prev = i32::MAX;
	let mut count = 0;
	let mut window = vec![];

	for line in input.split('\n') {
		let i: i32 = line.parse().unwrap();

		window.push(i);

		if window.len() >= 3 {
			let sum = window.iter().sum();
			//println!("sum {}", sum);

			if sum > prev {
				count += 1;
			}

			prev = sum;
			window.remove(0);
		}
	}

	return count.to_string()
}

pub const PUZZLE_DATA: aoc::Puzzle = aoc::Puzzle {
	day: 1,
	input: "file:01-input",
	implementation_part1: part1,
	implementation_part2: part2,
	tests_part1: &[
		("7", "file:01-input-test"),
	],
	tests_part2: &[
		("5", "file:01-input-test"),
	],
};

File: 2022/day01.py
from tools.aoc import AOCDay
from typing import Any

class Day(AOCDay):
	inputs = [
		[
			(24_000, '01-test'),
			(72_070, '01')
		],
		[
			(45_000, '01-test'),
			(211_805, '01')
		]
	]

	def part1(self) -> Any:
		return max(elfcal(self.input))

	def part2(self) -> Any:
		e = elfcal(self.input)
		e.sort(reverse = True)
		return sum(e[0:3])

def elfcal(input):
	elfcal = [0]

	for line in input:
		try:
			v = int(line)
			elfcal[-1] += v
		except:
			elfcal.append(0)

	return elfcal

